### ISSUE TITLE
Raspberry Pi 4 webserver issues.

### ISSUE THREAD

**Issue by saul-avikar** [nodogsplash/nodogsplash#548]
After installing nodogsplash it appears to not be working, using the default config file (with the only change being to the debugging number) I get this from the logread command:
```
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: maxclients, arg: 250
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: gatewayname, arg: OpenWrt Nodogsplash
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: gatewayinterface, arg: br-lan
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: gatewayport, arg: 2051
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: sessiontimeout, arg: 1200
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: preauthidletimeout, arg: 30
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: authidletimeout, arg: 120
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: checkinterval, arg: 600
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: FirewallRuleSet, arg: authenticated-users {
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing FirewallRuleSet authenticated-users
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: p1 = [FirewallRule]; p2 = [allow all]
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Adding FirewallRule allow all port (null) to 0.0.0.0/0 to FirewallRuleset authenticated-users
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: FirewallRuleSet authenticated-users parsed.
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: FirewallRuleSet, arg: preauthenticated-users {
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing FirewallRuleSet preauthenticated-users
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: FirewallRuleSet preauthenticated-users parsed.
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: FirewallRuleSet, arg: users-to-router {
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing FirewallRuleSet users-to-router
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: p1 = [FirewallRule]; p2 = [allow tcp port 22]
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Adding FirewallRule allow tcp port 22 to 0.0.0.0/0 to FirewallRuleset users-to-router
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: p1 = [FirewallRule]; p2 = [allow tcp port 23]
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Adding FirewallRule allow tcp port 23 to 0.0.0.0/0 to FirewallRuleset users-to-router
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: p1 = [FirewallRule]; p2 = [allow tcp port 53]
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Adding FirewallRule allow tcp port 53 to 0.0.0.0/0 to FirewallRuleset users-to-router
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: p1 = [FirewallRule]; p2 = [allow udp port 53]
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Adding FirewallRule allow udp port 53 to 0.0.0.0/0 to FirewallRuleset users-to-router
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: p1 = [FirewallRule]; p2 = [allow udp port 67]
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Adding FirewallRule allow udp port 67 to 0.0.0.0/0 to FirewallRuleset users-to-router
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: p1 = [FirewallRule]; p2 = [allow tcp port 80]
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Adding FirewallRule allow tcp port 80 to 0.0.0.0/0 to FirewallRuleset users-to-router
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: p1 = [FirewallRule]; p2 = [allow tcp port 443]
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Adding FirewallRule allow tcp port 443 to 0.0.0.0/0 to FirewallRuleset users-to-router
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: FirewallRuleSet users-to-router parsed.
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: FirewallRuleSet, arg: trusted-users {
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing FirewallRuleSet trusted-users
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: FirewallRuleSet trusted-users parsed.
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing option: FirewallRuleSet, arg: trusted-users-to-router {
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Parsing FirewallRuleSet trusted-users-to-router
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: FirewallRuleSet trusted-users-to-router parsed.
Fri Mar 13 03:53:49 2020 daemon.info nodogsplash[2559]: Done reading configuration file '/tmp/etc/nodogsplash_cfg015847.conf'
Fri Mar 13 03:53:49 2020 daemon.info nodogsplash[2559]: Initializing signal handlers
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Setting SIGCHLD handler to sigchld_handler()
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Setting SIGPIPE  handler to SIG_IGN
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Setting SIGTERM, SIGQUIT, SIGINT  handlers to termination_handler()
Fri Mar 13 03:53:49 2020 daemon.info nodogsplash[2559]: Operating System Uptime: 2287 seconds
Fri Mar 13 03:53:49 2020 daemon.info nodogsplash[2559]: main: System Uptime is 2287 seconds
Fri Mar 13 03:53:49 2020 daemon.info nodogsplash[2559]: Setting started_time
Fri Mar 13 03:53:49 2020 daemon.debug nodogsplash[2559]: Finding IP address of br-lan
Fri Mar 13 03:53:49 2020 daemon.notice nodogsplash[2559]: Detected gateway br-lan at 192.168.1.2 (dc:a6:32:61:fd:4c)
Fri Mar 13 03:53:49 2020 daemon.err nodogsplash[2559]: Could not create web server: Invalid argument
```

My network config /etc/config/network:
```

config interface 'loopback'
        option ifname 'lo'
        option proto 'static'
        option ipaddr '127.0.0.1'
        option netmask '255.0.0.0'

config globals 'globals'
        option ula_prefix 'fd1c:0dbc:7a07::/48'

config interface 'wan'
        option ifname 'eth1'
        option proto 'dhcp'

config interface 'lan'
        option ifname 'eth0'
        option proto 'static'
        option ipaddr '192.168.1.2'
        option netmask '255.255.255.0'
        option ip6assign '60'
        list dns '122.56.237.1'
        option gateway '192.168.1.254'
        option type 'bridge'
```
Firmware Version: `OpenWrt SNAPSHOT r12230-5715b21f80 / LuCI Master git-20.062.71982-95804e5`<br>
Kernel Version `4.19.101`<br>
This is Nodogsplash version `4.4.0`<br>
<br>
I cannot seem to figure out why it is throwing this error, any help point me into the right direction would be much appreciated.

**Comment by bluewavenet** [comment]
@saul-avikar 
This is a known problem with NDS v4.4.0 on some devices.
It is fixed in release 4.5.0
Unfortunately the package for this new release is not yet available on OpenWrt.
As you have a Rpi4, for now, you have 2 options:
 * downgrade to NDS v4.0.1
 * compile from the release source code using the OpenWrt SDK

Download NDS version 4.0.1 OpenWrt Package:
https://downloads.openwrt.org/releases/19.07.2/packages/aarch64_cortex-a53/routing/nodogsplash_4.0.1-1_aarch64_cortex-a53.ipk

Download NDS v4.5.0 source code:
https://github.com/nodogsplash/nodogsplash/archive/v4.5.0.tar.gz

**Comment by saul-avikar** [comment]
Thanks for the response.

Both the solutions you provided fix the problem. 
However the aarch64_cortex-a53 package did not work for me, I had to use the aarch64_cortex-a72:
https://downloads.openwrt.org/releases/19.07.2/packages/aarch64_cortex-a72/routing/nodogsplash_4.0.1-1_aarch64_cortex-a72.ipk
